liquidity-rewards/contracts/UniswapPoolRewards.sol

```diff
@@ -1,5 +1,193 @@
 // SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
 
-contract UniswapPoolRewards {
-  
-}
+import "./Dependencies/LPTokenWrapper.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+
+contract UniswapPoolRewards is LPTokenWrapper, Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    string constant public NAME = "UniswapPoolRewards";
+
+    IERC20 public elimuToken;
+
+    uint256 public duration;
+    uint256 public periodFinish = 0;
+    uint256 public rewardPeriod = 0;
+
+    uint256 public rewardRate = 0;
+    uint256 public lastUpdateTime;
+    uint256 public rewardPerTokenStored;
+    uint256 public totalRewardPaid = 0;
+    uint256 public periodRewardPaid = 0;
```

Comment:
If we move away from the concept of "reward period", I think we can delete this variable?